src/nspawn/nspawn-network.c

```diff
@@ -84,96 +84,95 @@ static int generate_mac(
         return 0;
 }
 
-int setup_veth(const char *machine_name,
-               pid_t pid,
-               char iface_name[IFNAMSIZ],
-               bool bridge) {
+int setup_veth(const char *machine_name, pid_t pid, char **veth) {
 
         _cleanup_netlink_message_unref_ sd_netlink_message *m = NULL;
```

Comment:
I think you leak memory when multiple --network-veth options are used, because sd_rtnl_message_new_link doesn't unref the existing message before storing the new one in the pointer.

This would need to be moved into the loop.